tests/generation/test_utils.py

```diff
@@ -1607,6 +1667,56 @@ def test_generate_continue_from_past_key_values(self):
                         )
                     )
 
+    def test_generate_with_and_without_position_ids(self):
+        for model_class in self.all_generative_model_classes:
+            config, input_ids, attention_mask, _ = self._get_input_ids_and_config()
+            model = model_class(config).to(torch_device).eval()
+            model_forward_args = inspect.signature(model.forward).parameters
+            if "position_ids" not in model_forward_args:
+                self.skipTest("This model doesn't use `position_ids`")
+
+            out_wo_positions = model.generate(input_ids, attention_mask=attention_mask, max_new_tokens=5)
+
+            # infer position ids from attn mask and generate again
+            position_ids = attention_mask.long().cumsum(-1) - 1
+            position_ids.masked_fill_(attention_mask == 0, 1)
+            position_ids = position_ids[..., -input_ids.shape[-1] :].view(-1, input_ids.shape[-1])
```

Comment:
Use the newly created function instead? The main goal of this test is to ensure we can pass `position_ids`, reusing the function can save us future trouble here :)

(a small unit test with numerical checks outside the test mixins for the newly created function would also be nice!)